lib/ansible/modules/network/eos/eos_interfaces.py

```diff
@@ -0,0 +1,292 @@
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Merge provided configuration with device configuration
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+        enabled: True
+      - name: Ethernet2
+        description: 'Configured by Ansible'
+        enable: False
+    state: merged
+
+# After state:
+# ------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+#    description "Configured by Ansible"
+#    shutdown
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+# Using replaced
+
+# Before state:
+# -------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Replaces device configuration of listed interfaces with provided configuration
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+        enabled: True
+      - name: Ethernet2
+        description: 'Configured by Ansible'
+        enabled: False
+    state: replaced
+
+# After state:
+# ------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+# !
+# interface Ethernet2
+#    description "Configured by Ansible"
+#    shutdown
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+# Using overridden
+
+# Before state:
+# -------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Overrides all device configuration with provided configuration
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+        enabled: True
+      - name: Ethernet2
+        description: 'Configured by Ansible'
+        enabled: False
+    state: overridden
+
+# After state:
+# ------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+# !
+# interface Ethernet2
+#    description "Configured by Ansible"
+#    shutdown
+# !
+# interface Management1
+#    ip address dhcp
+# !
+
+# Using deleted
+
+# Before state:
+# -------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+#    description "Interface 1"
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
+
+- name: Delete or return interface parameters to default settings
+  eos_interfaces:
+    config:
+      - name: Ethernet1
+    state: deleted
+
+# After state:
+# ------------
+#
+# veos#show running-config | section interface
+# interface Ethernet1
+# !
+# interface Ethernet2
+# !
+# interface Management1
+#    description "Management interface"
+#    ip address dhcp
+# !
```

Comment:
IMO it's better to include example for delete w/o config passed as that'll give user better idea of which `delete` pattern to use

Reply:
I'm not sure I follow. Are you suggesting just `config:`? Because that will knock the device offline by removing the Management IP address. For `eos_interfaces` specifically, empty config delete should almost certainly never be used.